extensions/ql-vscode/src/qltest-discovery.ts

```diff
@@ -146,70 +162,43 @@ export class QLTestDiscovery extends Discovery<QLTestDiscoveryResults> {
   }
 
   protected async discover(): Promise<QLTestDiscoveryResults> {
-    const testDirectories: QLTestDirectory[] = [];
-    const watchPaths: string[] = [];
-    const qlPacks = this.qlPackDiscovery.qlPacks;
-    for (const qlPack of qlPacks) {
-      //HACK: Assume that only QL packs whose name ends with '-tests' contain tests.
-      if (this.isRelevantQlPack(qlPack)) {
-        watchPaths.push(qlPack.uri.fsPath);
-        const testPackage = await this.discoverTests(qlPack.uri.fsPath, qlPack.name);
-        if (testPackage !== undefined) {
-          testDirectories.push(testPackage);
-        }
-      }
-    }
-
-    return { testDirectories, watchPaths };
+    return {
+      testDirectory: await this.discoverTests(),
```

Comment:
We now unconditionally run test discovery on all workspace folders. How does this scale/perform when we have a large workspace with many folders, or a single large repo (like our internal codebase)? Does it also attempt to search the source archives that we add to the workspace in the extension, and if so can we skip them?